app/src/component/emailRewardSubcard/view.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+import React from 'react';
+import {
+  ActivityIndicator,
+  AsyncStorage,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View
+} from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome5';
+import Button from '../button';
+import Colors from '../../styles/colors';
+import Constants from '../../constants';
+import Link from '../link';
+import rewardStyle from '../../styles/reward';
+
+class EmailRewardSubcard extends React.PureComponent {
+  state = {
+    email: null,
+    verfiyStarted: false
+  };
+
+  componentDidMount() {
+    const { emailToVerify } = this.props;
+    AsyncStorage.getItem(Constants.KEY_FIRST_RUN_EMAIL).then(email => {
+      if (email && email.trim().length > 0) {
+        this.setState({ email });
+      } else {
+        this.setState({ email: emailToVerify });
+      }
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
No more legacy methods!
https://github.com/lbryio/lbry-android/pull/232/files#r210515192

Reply:
See the big yellow box:
https://reactjs.org/docs/react-component.html#updating

Reply:
Heh! The reason I haven't moved on from this yet is we're still using an older React version, 16.2.0. I've read the documentation and it looks like the warnings started appearing from v16.3.0. I haven't had the chance to explore upgrading to newer versions of react yet (there were some issues the last time I tried), but I would love to discard all legacy methods when the time comes.